### ISSUE TITLE
Do a complete documentation of all possible options for each plugins

### ISSUE THREAD

**Issue by svandecappelle** [svandecappelle/jQPlot#10]

**Comment by gelo32k** [comment]
Hello, I want to use your version of jQPlot since it is the most actively developed (The original jqPlot has its last commit 2 years ago). however, with the many commits that you have made I am now wondering. Does the documentation on the old jqplot is still applicable to your version of jQPlot?

**Comment by svandecappelle** [comment]
Yes it is still applicable, I just added some options and features. But all the original existing options are still applicable.

**Comment by gelo32k** [comment]
Ok thanks, I'll be using this :)

**Comment by johanbove** [comment]
Would this be a migration of the existing documentation over on the website?
@svandecappelle Do you have an overview of the additions you added?

**Comment by svandecappelle** [comment]
@johanbove There is no documentation page that show the additions from the original sources. But: 

* This is essentially bug fixes.
Modifications affects essentially:
* donutRenderer / pieRenderer (colors and hover improments)
* pointsLabel (bugs fixes)1
* jsDate formatter improvments
* Support of filling shapes of line and bar renderer by a bi-color
gradient
* add -> stairsRenderer
* canvasOverlay can use date axes
* Export to image bug fixes
* allow a global background color to canvas 
* bug fixes in tooltip
* adding shape options support to vertical & horizontal line on cursor 
* color series support in cursor highlighter
* format string to cursor

In the documentation i will highlight the new features.

**Comment by RubenOlsen** [comment]
I'll start the work on generating API docs based on the current sources. Ref the mails on the list - the generated API documentation will be available in the GH-pages.

I think the first thing that needs to be documented is the API, secondly the examples.

**Comment by svandecappelle** [comment]
I Have not yet worked on API documentation generation. It looks cool to use jekyll for atuomatic generation.
The site is not generated. I all write it by hand. But the API documentation could be generated.

I wanted the site contains:
* The main page presentation
* The API documentation (Generated by jekyll)
* The demonstration
* The Unit tests
* The plugins contribution / presentation documentation
* Usefull and downloadable links

**Comment by RubenOlsen** [comment]
Very good.

I'll keep the structure and everything. It's a good structure for what we need with regards to our documentation.

Jekyll is not used for generating the API documentation - jsdoc (http://usejsdoc.org/) is a good choice. 

Jekyll is a good website builder which uses templates and stuff. This means that if we decide to add a new menu-choice, or change the content of the <head> ... </head> sections, etc - this will be automatically changed on all webpages without having to do any global search&replace. Using a tool like jekyll means that we'll never forget to add "common stuff" to new pages.

I'll do a pull-request on the gh-pages branch some time during this weekend.

Could you also change my status in the project so that i can assign tickets to myself?

**Comment by johanbove** [comment]
:+1: Thanks for the commitment Ruben!
